src/lib/IlmThread/IlmThreadPool.cpp

```diff
@@ -254,7 +223,7 @@ DefaultWorkerThread::run ()
         _data->taskSemaphore.wait();
 
         {
-            Lock taskLock (_data->taskMutex);
+            std::unique_lock<std::mutex> taskLock (_data->taskMutex);
```

Comment:
Why unique_lock here but lock_guard in other places?

Reply:
oh, because this location does an explicit unlock to start the task. I could put it in a scope if that'd be clearer?

Reply:
No, I was just curious.